src/Database/Driver.php

```diff
@@ -426,6 +427,18 @@ public function newCompiler()
         return new QueryCompiler();
     }
 
+    /**
+     * Constructs new TableSchema.
+     *
+     * @param string $table The table name.
+     * @param array $columns The list of columns for the schema.
+     * @return \Cake\Database\Schema\TableSchema
+     */
+    public function newTableSchema($table, $columns = [])
+    {
+        return new TableSchema($table, $columns);
```

Comment:
You could check for presence of `$_config['tableSchema']` key and if present use that as class name else fallback to `TableSchema`. That would allow users to easily use their custom class without having to override this method, as you proposed in #13689.

Reply:
That I do not understand. For example in TestFixture.php https://github.com/cakephp/cakephp/pull/13696/files#diff-56561cb12d6a5162f76fe28b07514e1cL182 we have `$this->_schema = $connection->getDriver()->newTableSchema($this->table);` now. How would users use their custom table schema class there without overriding `newTableSchema`?

Reply:
They would set the `tableSchema` key in connection key. That doesn't change whether it's your production connection or test suite connection.

Reply:
Do you mean the same what I wrote in the [issue](https://github.com/cakephp/cakephp/issues/13689)?
> One option would be to set it like connection and database driver classes are set for ConnectionManager::setConfig method:

    'className' => Connection::class,
    'driver' => Mysql::class,
    'tableSchemaClass' => TableSchema::class,

@markstory suggested the second option with the getter:
> The method on the connection sounds like a good approach to me. 

Or do you suggest both?

Reply:
```php
$className = $this->_config['tableSchema'] ?? TableSchema::class;

return new $className($table, $columns);
```